File: app/src/main/java/org/deiverbum/app/data/entity/relation/OficceVerseAll.java
package org.deiverbum.app.data.entity.relation;

import androidx.room.Embedded;
import androidx.room.Relation;

import org.deiverbum.app.data.entity.LHOficceVerseEntity;
import org.deiverbum.app.data.entity.LHOficceVerseJoinEntity;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2022.2
 */
public class OficceVerseAll {
    @Embedded
    public LHOficceVerseJoinEntity theJoin;
    @Relation(
            parentColumn = "verseFK",
            entityColumn = "verseID",
            entity = LHOficceVerseEntity.class
    )
    public LHOficceVerseEntity theEntity;

    public String getDomainModel(){
        return theEntity.getResponsorio();
    }

}

File: app/src/main/java/org/deiverbum/app/data/entity/relation/OficceVerseAll.kt
package org.deiverbum.app.data.entity.relation

import androidx.room.Embedded
import androidx.room.Relation
import org.deiverbum.app.data.entity.LHOfficeVerseEntity
import org.deiverbum.app.data.entity.LHOfficeVerseJoinEntity

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
data class OficceVerseAll(
    @Embedded
    var theJoin: LHOfficeVerseJoinEntity,

    @Relation(
        parentColumn = "verseFK",
        entityColumn = "verseID",
        entity = LHOfficeVerseEntity::class
    )
    var theEntity: LHOfficeVerseEntity
)
